app/gen-server/lib/HomeDBManager.ts

```diff
@@ -2479,6 +2490,85 @@ export class HomeDBManager extends EventEmitter {
       .getOne() || undefined;
   }
 
+  // can be factorized with getShareByLinkId
+  public async getDocApiKeyByLinkId(docId: string, linkId: string): Promise<Share | undefined> {
+    return await this._connection.createQueryBuilder()
+      .select('shares')
+      .from(Share, 'shares')
+      .where('shares.doc_id = :docId', {docId})
+        .andWhere('shares.link_id = :linkId', {linkId})
+      .getOne() || undefined;
+  }
+
+  public async getDocApiKeyByKey(key: string): Promise<Share | null> {
+    return await this.getShareByKey(key);
+  }
+
+  public async createDocApiKey(docId: string, share: ShareInfo) {
+    const key = makeId();
+    const query = await this._connection.createQueryBuilder()
+      .insert()
+      .setParameter('options', share.options)
+      .into(Share)
+      .values({
+        linkId: share.linkId,
+        docId: docId,
+        options: JSON.parse(share.options),
+        key,
+      })
+      .execute() || undefined;
+    return query ? key : query;
+  }
+
+  // in parameters linkId is the linkId in db in case of update of this id in the share
```

Comment:
I took some time to understand this comment, what do you think of this instead:

```suggestion
  // the linkId parameter corresponds to the one currently stored in the db in case we want to change it.
```

Reply:
In fact any parameter can be updated, not just linkId.
If we want to update linkId it have to be set in share object passed as a parameter
`share.linkId = newLinkId`

So I go with

```
  // nb. The linkId parameter corresponds to the one currently stored in the db.
  // If we want to update it the new value must be passed through the share parameter
```

Let me know